### PR TITLE
Add command-line-api repo reference

### LINKED ISSUE #42637: Missing command-line-api repo reference in eng/Version.Details.xml
Runtime uses the dotnet/command-line-api repo, e.g. [here](https://github.com/dotnet/runtime/blob/57fde8309798a53c4bf6eb2b579683d9f3e71bf6/src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj#L38), but this is not included in eng/Version.Details.xml, so Arcade/Maestro package updates and uberclone will not know about the dependency.  This should be added at an appropriate version and hash.

### PR DESCRIPTION
**Pull request by AntonLapounov**
Fixes #42637.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -90,6 +90,10 @@
       <Uri>https://github.com/microsoft/vstest</Uri>
       <Sha>2418d9ec492e93245e1ce8699d6ae9cd7b86649b</Sha>
     </Dependency>
+    <Dependency Name="System.CommandLine" Version="2.0.0-beta1.20253.1" Pinned="true">
```
Comment (am11) [inline eng/Version.Details.xml:93]: This is also listed at: https://github.com/dotnet/runtime/blob/7e1ad2795b4c8f6b0085ddbbb477c807e587cca4/eng/Versions.props#L124 can we consolidate dependency versions in one file?

**Comment by jkotas** [comment]
System.CommandLine package has poor performance that we need to do something about. Also, I suspect that System.CommandLine will require more work in https://github.com/dotnet/source-build .

Should we rather switch back to the light-weight command line parser we used to have? (You can take it from https://github.com/dotnet/runtimelab/tree/feature/NativeAOT/src/coreclr/src/tools/System.CommandLine/System/CommandLine).

**Comment by ViktorHofer** [comment]
cc @trylek who I believe switched to it (https://github.com/dotnet/runtime/commit/50c3665484f5b286d311bf432e0497699ff2b406).

**Comment by crummel** [comment]
@jkotas We actually already added this as part of our prebuilt removal - I logged this issue because System.CommandLine was not actually included in the tree anywhere and we had to add it to our Version.Details.xml.  That being said, source-build will never be upset about removing a repo dependency if that works better for you.

**Review by ViktorHofer — APPROVED** [review]

**Comment by AntonLapounov** [comment]
According to @crummel, we only need to remove this dependency from shipping components and anything that is required to build those shipping components. I am abandoning this PR.

**Pull request closed without merging**